azure-templates/deploy-steps.yml

```diff
@@ -0,0 +1,128 @@
+# Azure DevOps template used to perform all the deploy steps including build. 
+#
+# Note. Deployment slots let you deploy different versions of your function
+# app to different URLs. You can test a certain version and then swap content
+# and configuration between slots to have minimal impact to production and also 
+# make rollback easily.
+
+parameters:
+  - name: 'deployType'
+    type: string
+    default: deployToProductionSlot
+    values:
+    - deployToProductionSlot
+    - deployToStagingSlotAndSwap
+    - deployToStagingSlot  
+    - deployToTestSlot  
+  
+  - name: 'azureSubscription'
+    type: string
+    default: ''
+  
+  - name: 'resourceGroupName'
+    type: string
+    default: ''
+  
+  - name: 'appName'
+    type: string
+    default: ''
+    
+steps:
+  - template: ./make-build-steps.yml
+    parameters:
+      make: predeploy_build
+
+  # Install functions extensions
+  - task: DotNetCoreCLI@2
+    inputs:
+      command: "build"
+      arguments: "-o bin"
+  
+  - task: CopyFiles@2
+    inputs:
+      SourceFolder: '$(System.DefaultWorkingDirectory)'
+      TargetFolder: '$(Build.ArtifactStagingDirectory)'
+      Contents: |
+        **/*
+        !.git/**/*
+        !**/*.js.map
+        !**/*.ts
+        !.vscode/**/*
+        !azure-templates/**/*
+        !azure-pipelines.yml
+        !.prettierrc
+        !.gitignore
+        !README.md
+        !jest.config.js
+        !local.settings.json
+        !test
+        !tsconfig.json
+        !tslint.json
+        !yarn.lock
+        !Dangerfile.js
+        !CODEOWNERS
+        !__*/**/*
+    displayName: 'Copy deploy files'
+  
+  # Option 1: standard deployment without slots (i.e. deploy directly to main slot)
+  - ${{ if eq(parameters.deployType, 'deployToProductionSlot') }}:
+    - task: AzureFunctionApp@1
+      inputs:
+        azureSubscription: '${{ parameters.azureSubscription }}'
+        resourceGroupName: '${{ parameters.resourceGroupName }}'
+        appType: 'functionApp'
+        appName: '${{ parameters.appName }}'
+        package: '$(Build.ArtifactStagingDirectory)/'
+        deploymentMethod: 'auto'
+      displayName: Deploy to production slot
+  
+  # Option 2: deployment to 'staging' slot only
+  - ${{ if eq(parameters.deployType, 'deployToStagingSlot') }}:
```

Comment:
At the moment the infrastructure only has a production slot for this function

Reply:
The deploy is set to production slot

https://github.com/pagopa/io-functions-backoffice/blob/175573029-add-azure-pipeline/azure-pipelines.yml#L66

https://github.com/pagopa/io-functions-backoffice/blob/175573029-add-azure-pipeline/azure-templates/deploy-steps.yml#L11

Reply:
yes, but could be it manually selected?

Reply:
Correct choice is preselected, but you can manual select different choice but they'll fail.

All others azure devops pipelines have the same scaffolder and it is the reason because you see all the options.

Maybe we can remove `deployToStagingSlot`and `deployToStagingSlotAndSwap` here
https://github.com/pagopa/io-functions-backoffice/blob/175573029-add-azure-pipeline/azure-pipelines.yml#L67

Reply:
Agree, so we have aligned this pipeline with the real infrastructure (even for documentation purposes).

Reply:
ok removed (commented) `deployToStagingSlotand` `deployToStagingSlotAndSwap`